Assets/MixedRealityToolkit.Providers/ObjectMeshObserver/SpatialObjectMeshObserverProfileInspector.cs

```diff
@@ -0,0 +1,145 @@
+
+        // General settings
+        private SerializedProperty startupBehavior;
+        private SerializedProperty observationExtents;
+        private SerializedProperty observerVolumeType;
+        private SerializedProperty isStationaryObserver;
+        private SerializedProperty updateInterval;
+
+        // Physics settings
+        private SerializedProperty meshPhysicsLayer;
+        private SerializedProperty recalculateNormals;
+
+        // Level of Detail settings
+        private SerializedProperty levelOfDetail;
+        private SerializedProperty trianglesPerCubicMeter;
+
+        // Display settings
+        private SerializedProperty displayOption;
+        private SerializedProperty visibleMaterial;
+        private SerializedProperty occlusionMaterial;
+
+        private readonly GUIContent displayOptionContent = new GUIContent("Display Option");
+        private readonly GUIContent lodContent = new GUIContent("Level of Detail");
+        private readonly GUIContent volumeTypeContent = new GUIContent("Observer Shape");
+        private readonly GUIContent physicsLayerContent = new GUIContent("Physics Layer");
+        private readonly GUIContent trianglesPerCubicMeterContent = new GUIContent("Triangles/Cubic Meter");
+
+        protected override void OnEnable()
+        {
+            base.OnEnable();
+
+            // Object mesh
+            spatialMeshObject = serializedObject.FindProperty("spatialMeshObject");
+
+            // General settings
+            startupBehavior = serializedObject.FindProperty("startupBehavior");
+            observationExtents = serializedObject.FindProperty("observationExtents");
+            observerVolumeType = serializedObject.FindProperty("observerVolumeType");
+            isStationaryObserver = serializedObject.FindProperty("isStationaryObserver");
+            updateInterval = serializedObject.FindProperty("updateInterval");
+
+            // Mesh settings
+            meshPhysicsLayer = serializedObject.FindProperty("meshPhysicsLayer");
+            recalculateNormals = serializedObject.FindProperty("recalculateNormals");
+            levelOfDetail = serializedObject.FindProperty("levelOfDetail");
+            trianglesPerCubicMeter = serializedObject.FindProperty("trianglesPerCubicMeter");
+            displayOption = serializedObject.FindProperty("displayOption");
+            visibleMaterial = serializedObject.FindProperty("visibleMaterial");
+            occlusionMaterial = serializedObject.FindProperty("occlusionMaterial");
+        }
+
+        public override void OnInspectorGUI()
+        {
+            RenderProfileHeader(ProfileTitle, ProfileDescription, target, true, BackProfileType.SpatialAwareness);
+
+            using (new GUIEnabledWrapper(!IsProfileLock((BaseMixedRealityProfile)target)))
+            {
+                serializedObject.Update();
+
+                EditorGUILayout.PropertyField(spatialMeshObject);
```

Comment:
The provider really doesn't do anything if spatialMeshObject doesn't have at least one MeshFilter. Therefore it probably makes sense for the inspector to warn the user if they select a GameObject here that doesn't have any MeshFilters using **GetComponentsInChildren**.

Reply:
i like that